pkg/download/download.go

```diff
@@ -22,20 +22,21 @@ import (
 
 const b64 = "base64"
 
-type downloadOptions struct {
+type DownloadOptions struct {
```

Comment:
I don't think we need to have this public, we could rather do an option which provides a function which returns the endpoint.

Something like `WithEndpointFunc(func(ctx context.Context) (string, error)`

Reply:
So unless `WithEndpointFunc()` is specified, we use the `endpoint` arg to keep backwards compat